From: SeongJae Park <sj@kernel.org>
Date: Sun, 15 Feb 2026 12:51:39 -0800
Subject: [PATCH] samples/damon/mtier: replace damon_add_region() with
 damon_set_regions()

We want to hide damon_add_region().

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 samples/damon/mtier.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/samples/damon/mtier.c b/samples/damon/mtier.c
index 775838a23d935..b84f991c8b9b7 100644
--- a/samples/damon/mtier.c
+++ b/samples/damon/mtier.c
@@ -80,6 +80,7 @@ static struct damon_ctx *damon_sample_mtier_build_ctx(bool promote)
 	struct damos_quota_goal *quota_goal;
 	struct damos_filter *filter;
 	struct region_range addr;
+	struct damon_addr_range range;
 	int ret;
 
 	ctx = damon_new_ctx();
@@ -120,10 +121,12 @@ static struct damon_ctx *damon_sample_mtier_build_ctx(bool promote)
 		addr.end = promote ? node1_end_addr : node0_end_addr;
 	}
 
-	region = damon_new_region(addr.start, addr.end);
-	if (!region)
+	range.start = addr.start;
+	range.end = addr.end;
+
+	ret = damon_set_regions(target, &range, 1, DAMON_MIN_REGION_SZ);
+	if (ret)
 		goto free_out;
-	damon_add_region(region, target);
 
 	scheme = damon_new_scheme(
 			/* access pattern */
-- 
2.47.3

